### PR TITLE
ticket7 tested with error handling

### PR DESCRIPTION
**Pull request by jd4545**
Tested functionality and 400/404 errors

### CODE REVIEW

**Comment by jd4545** [comment]
Have noticed that I've not tested for
-- No inc_votes on request body
-- Invalid inc_votes (e.g. { inc_votes : "cat" }) 
Will do this and resubmit pull request, so please ignore for now.

File: models.js [hunk 1]
```diff
@@ -17,3 +17,23 @@ exports.fetchArticleByID = (park_id) => {
         return rows[0];
     })
 };
+
+exports.updateArticleById = (article_id, articleInfo) => {
+    return db.query(`SELECT votes FROM articles WHERE article_id = $1;`, [article_id])
+    .then(({rows}) => {
+        if (rows.length === 0) {
+            return Promise.reject({ status: 404, msg: "Article not found"})
+        } else {
```
Comment (daviidbartlett) [inline models.js]: you don't need to select the article first. What happens if you try to update an article which doesn't exist? Please refactor!

**Review by daviidbartlett — COMMENTED** [review]
model needs a refactor

File: models.js (hunk 1 above)
Comment (jd4545) [reply models.js]: Hi David thanks for the feedback.

I introduced the initial article select for those cases where the seeded votes are non-zero. My understanding of the ticket is that I should increment/decrement the vote field by the amount set by the client. So the select allows me to look into the db to do this.

Regarding the error - I get a TypeError - cannot read properties of undefined (reading 'votes'). I'm unclear as to what to do about this part.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -76,4 +76,139 @@ describe("News app",()=>{
...
+        })
+    })
+    test('status:400, incorrect format in req.body', () => {
+        const articleUpdate = {inc_votes: "What's up pussycat, Whoaoaoaoh" };
+        return request(app)
+            .patch('/api/articles/1')
+            .send(articleUpdate)
+            .expect(400)
+            .then(({ body }) => {
+                expect(body.msg).toBe("Invalid id entered")
```
Comment (Poonam-raj) [inline __tests__/app.test.js]: This error message is incorrect for what is happening here - I think something relating to 'bad request' would be better to suggest to the user that they've made that body badly

File: models.js [hunk 3]
```diff
@@ -17,3 +17,26 @@ exports.fetchArticleByID = (park_id) => {
         return rows[0];
     })
 };
+
+exports.updateArticleById = (article_id, articleInfo) => {
```
Comment (Poonam-raj) [inline models.js:21]: This model looks better

**Review by Poonam-raj — CHANGES_REQUESTED** [review]
Just want you to amend that error message that doesn't reflect the scenario you're testing in but otherwise looks good.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -145,7 +145,7 @@ describe("News app",()=>{
         return request(app).patch('/api/articles/Bad-id')
```
Comment (cjplatten) [inline __tests__/app.test.js:145]: The test name needs updating to reflect the new message

**Review by cjplatten — COMMENTED** [review]
One thing to address but otherwise this is good to go!

File: __tests__/app.test.js (hunk 4 above)
Comment (jd4545) [reply __tests__/app.test.js:145]: Thanks, will update and merge

**Pull request merged**